### PR TITLE
Store osl version locally

### PR DESCRIPTION
**Pull request by PanekOndrej**
Some functionality dependent on optiSLang version is planned (e. g. `connect_nodes` in version > 24.1 will use server command for connection of inner slots instead of API script). 

- This will reduce unnecessary communication with the server when checking for version.

### CODE REVIEW

**Comment by PanekOndrej** [comment]
Please wait a while with the merge, I have to modify unit tests and methods in `Optislang` class.

File: src/ansys/optislang/core/tcp_osl_server.py [hunk 1]
```diff
@@ -1063,16 +1065,39 @@ def __init__(
...
+        if self.__osl_version[0] is not None:
+            if self.__osl_version[0] < 23:
                 self._logger.warning(
-                    f"The version of the used Ansys optiSLang ({osl_version_string})"
+                    f"The version of the used Ansys optiSLang ({self.__osl_version_string})"
                     " is not fully supported. Please use at least version 23.1."
                 )
 
+    @property
+    def osl_version(self) -> Tuple[Optional[int], ...]:
```
Comment (widerschein) [inline src/ansys/optislang/core/tcp_osl_server.py:1079]: Tuple length is known at this point.

**Comment by widerschein** [comment]
Some general thought on this functionality: The version string is pretty much guaranteed to be of the form
 `2X.X.Xyyy (N+)`
with `yyy` as an optional suffix. So a regular expression could grab the components accordingly. Failing to parse would then be a runtime error. This way only the suffix would be an optional value. Users of the library would not have to check each component for `None`.
It would also be very cool to have the result as a namedtuple.

**Comment by PanekOndrej** [comment]
> Some general thought on this functionality: The version string is pretty much guaranteed to be of the form `2X.X.Xyyy (N+)` with `yyy` as an optional suffix. So a regular expression could grab the components accordingly. Failing to parse would then be a runtime error. This way only the suffix would be an optional value. Users of the library would not have to check each component for `None`. It would also be very cool to have the result as a namedtuple.

Method `_get_osl_version()` returns tuple of length = 4 (major, minor, maint, rev). As I know, major and minor version should be always parsed correctly, I'm not sure about the other 2 (for example, this is what I get with currently installed osl: `(24, 1, None, 2342)`). 

So raise error only if **major** and **minor** version not parsed correctly, keep the other 2 as `Optional[int]` and implement like this?
```
from typing import NamedTuple

class OslVersion(NamedTuple):
    """optiSLang version as tuple containing major version, minor version, maintenance version and revision."""
    major: int
    minor: int
    maintenance: Optional[int]
    revision: Optional[int]
```

**Comment by widerschein** [comment]
>     * The revision version should be always parsed correctly?
I think it might be possible that the revision contains non-numerals. I have seen this in local builds with something like `(123M)`. Let me check this.

>     * Modify it so that `0` is extracted from the `'0dev'` part?
That's probably the best. Just use the numeral of the maintenance part and ignore any appended non-numerals (until the first whitespace or opening parens of the revision part).

**Comment by widerschein** [comment]
The revision component can have a single additional character as a suffix: `(123M)`. That happens in some cases and for internal builds of optiSLang. It will not be the case for installed versions.
For development purposes it would still be good to consider this case (and just remove the suffix).

**Comment by PanekOndrej** [comment]
I rather kept `maintenance` and `revision` versions as `Optional` since this won't be necessary for version checks, but it should be parsed correctly even in cases with `dev` or `M`.

**Pull request merged**